src/main/java/Duke.java

```diff
@@ -1,10 +1,258 @@
+    // Keep track of number of existing tasks in list
+    public static int inputCount = 0;
+    // FilePath
+    public static String filePath = "src\\main\\java\\duke.txt";
+    // Print greeting message upon startup of program
+    public static void printGreeting() {
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        // Greeting Message
+        System.out.println("Hello, I'm Duke!");
+        System.out.println("What can I do for you?");
+    }
+    // Function to label task as done
+    public static void doneTask(String task) throws NumberFormatException {
+        try {
+            int doneTask = Integer.parseInt(task.replaceAll("[\\D]",""));
+            list[doneTask - 1].markAsDone();
+            System.out.println("______________________________________________");
+            System.out.println("Done. I have marked this task as done:");
+            System.out.println("   " + list[doneTask - 1].toString());
+            System.out.println("______________________________________________");
+        } catch (NumberFormatException e) {
+            System.out.println("Error: Did not specify which task is completed!");
+        }
+    }
+    // Function to list all tasks stored
+    public static void listTasks() {
+        System.out.println("Here are the tasks in your list:");
+        System.out.println("______________________________________________");
+        for(int i = 0; i < inputCount; i++) {
+            System.out.println((i+1) + "." + list[i].toString());
+        }
+        System.out.println("______________________________________________");
+    }
+
+    public static void deleteTask(String delete) throws NumberFormatException, NullPointerException, ArrayIndexOutOfBoundsException {
+        try {
+            int toDelete = Integer.parseInt(delete.replaceAll("[\\D]", ""));
+            toDelete--;
+            String toPrint = list[toDelete].toString();
+            System.out.println("______________________________________________");
+            System.out.println("Noted. I've removed this task:");
+            System.out.println("   " + toPrint);
+            Task[] newList = new Task[MAX_TASKS];
+            for (int i = 0, j = 0; i < inputCount; i++) {
+                if (i == toDelete) {
+                    continue;
+                }
+                newList[j++] = list[i];
+            }
+            inputCount--;
+            list = newList;
+            System.out.println("Now you have " + inputCount + " tasks in the list.");
+            System.out.println("______________________________________________");
+        } catch (NumberFormatException e) {
+            System.out.println("Error: Did not specify which task to delete!");
```

Comment:
good that you have separated the types of exceptions, you can consider doing the same for the output statements (replace them with string constants which you can declare at the top)